Sources/NIOCore/ByteBuffer-hex.swift

```diff
@@ -341,3 +355,54 @@ extension ByteBuffer {
         }
     }
 }
+
+enum HexDecodingError: Error {
```

Comment:
Can we make this `public`, and make it a `struct`?

Reply:
Also needs to _either_

- be nested into `ByteBuffer`
- or have a `NIO` prefix as per the public API guarantees that we don't add top-level things that aren't `NIO`-prefixed